packages/firecamp-echo-server/src/rest/rest.controller.ts

```diff
@@ -135,50 +171,209 @@ export class RestController {
+        const decodedSplit = decoded.split(':')
+        if (decodedSplit.length != 2){
+            return res.status(401).send('Unauthorized')
+        }
+        const username = decodedSplit[0]
+        const password = decodedSplit[1]
+
+        if (username !== echo_username || password !== echo_password){
+            return res.status(401).send('Unauthorized')
+        }
+        
+        return { authenticated: true }
     }
 
     // Digest Auth
     @Get('digest-auth')
-    digestAuth() {
-        return 'yet to implement'
-        // return { authenticated: true }
+    digestAuth(@Req() req,@Headers() headers, @Response() res) {
+        const realm = 'Users'
+
+        if (!('authorization' in headers)){
+            const nonce = crypto.randomBytes(16).toString('base64');
+            res.set({'www-authenticate':`Digest realm="${realm}", nonce="${nonce}"`}).status(401);
+            return res.status(401).send('Unauthorized')
+        }
+
+        const split = headers.authorization.split(" ")
+        const type = split[0]
+        if (type !== 'Digest'){
+            return res.status(401).send('Unauthorized')
+        }
+       
+        
+        const authArgs = headers.authorization.slice(7).split(', ')
+        const argsMap = {}
+        
+        authArgs.forEach(arg => {
+            const split = arg.replaceAll('"','').replace('=', '-:-').split('-:-')
+            argsMap[split[0]] = split[1]
+        })
+        
+        if (!(argsMap['username'] && argsMap['nonce'] && argsMap['realm'] && argsMap['response'])){
+            return res.json({"authorized": false})
+        }
+        const HA1 = crypto.createHash('md5').update(`${argsMap['username']}:${argsMap['realm']}:${echo_password}`).digest('hex')
+        const HA2 = crypto.createHash('md5').update(`GET:/digest-auth`).digest('hex')
+        const responseCheck = crypto.createHash('md5').update(`${HA1}:${argsMap['nonce']}:${HA2}`).digest('hex')
+        
+        if (responseCheck !== argsMap['response']){
+            return res.status(401).send('Unauthorized')
+        }
+
+        return res.json({"authorized":true})
     }
 
     // Hawk Auth
     @Get('hawk-auth')
-    hawkAuth() {
-        return 'yet to implement'
-        // return { authenticated: true }
+    async hawkAuth(@Req() req, @Response() res) {
+        const credentialsFunc = function (id) {
+
+            if (id !=='dh37fgj492je'){
+                return undefined
+            }
+            
+            const credentials:Credentials = {
+                key: 'werxhqb98rpaxn39848xrunpaw3489ruxnpa98w4rxn',
+                algorithm: 'sha256',
+                user: 'Steve'
+            };
+        
+            return credentials;
+        };
+
+        let status, message
+
+        try {
+            const { credentials, artifacts } = await Hawk.server.authenticate(req, credentialsFunc)
+            message = {'message':"Hawk Authentication Successfull"}
```

Comment:
fwiw, there was a real spelling error hiding in the noise :)
```suggestion
            message = {'message':"Hawk Authentication Successful"}
```